packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/maps_object.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show objectRuntimeType;
+import 'package:meta/meta.dart' show immutable;
+
+/// Uniquely identifies object an among [GoogleMap] collections of a specific
+/// type.
+///
+/// This does not have to be globally unique, only unique among the collection.
+@immutable
+class MapsObjectId<T> {
+  /// Creates an immutable object representing a [T] among [GoogleMap] Ts.
+  ///
+  /// An [AssertionError] will be thrown if [value] is null.
+  MapsObjectId(this.value) : assert(value != null);
+
+  /// The value of the id.
+  final String value;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final MapsObjectId<T> typedOther = other as MapsObjectId<T>;
+    return value == typedOther.value;
+  }
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  String toString() {
+    return '${objectRuntimeType(this, 'MapsObjectId')}{value: $value}';
```

Comment:
Should we follow the style guide here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#override-tostring

Reply:
Done. There was inconsistency in () vs {} in the code I consolidated; I didn't realize there was style guidance on this so had just picked arbitrarily.